buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py

```diff
@@ -0,0 +1,131 @@
+"""Authors: Heberto Mayorquin and Cody Baker."""
+from pathlib import Path
+import warnings
+
+import numpy as np
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+from pynwb.file import NWBFile, TimeIntervals
+from pynwb.behavior import SpatialSeries, Position
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from nwb_conversion_tools.utils.conversion_tools import get_module
+from nwb_conversion_tools.utils.json_schema import FolderPathType
+
+from .tingleyseptal_utils import read_matlab_file
+
+
+class TingleySeptalBehaviorInterface(BaseDataInterface):
+    """Behavior data interface for the Tingley Septal project."""
+
+    def __init__(self, folder_path: FolderPathType):
+        super().__init__(folder_path=folder_path)
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+        session_path = Path(self.source_data["folder_path"])
+        session_id = session_path.stem
+
+        # Position
+        module_name = "Position"
+        module_description = "Contains behavioral data concerning position."
+        processing_module = get_module(nwbfile=nwbfile, name=module_name, description=module_description)
+
+        behavior_file_path = Path(session_path) / f"{session_id}.behavior.mat"
+
+        behavior_mat = read_matlab_file(str(behavior_file_path))["behavior"]
+        time_stamps = behavior_mat["timestamps"]
```

Comment:
Doesn't look like you use these below after extracting them. Are they regular? (`from nwb_conversion_tools.utils.conversion_tools import check_regular_timestamps`, or something similarly named)

If they're regular you can forego all of it and use `starting_time` and `rate` as usual, if not we should use the timestamps instead.

Reply:
I was actually waiting to discuss this with you but it seems that all sampling is non-uniform.